docs/development/testguide.rst

```diff
@@ -117,6 +100,27 @@ This can be used in conjunction with the ``-P`` option provided by the
 `pytest-filter-subpackage <https://github.com/astropy/pytest-filter-subpackage>`_
 plugin to run just part of the test suite.
 
+Using tox to set up a development environment
+---------------------------------------------
+
+As noted, running ``tox`` can be quite slow, yet it has the advantage over the
+other methods that it puts together a consistent set of dependencies.  One can
+speed up matters by using ``tox`` to set up a virtual environment for
+development and then use direct ``pytest`` commands within that environment.
+For instance::
+
+    tox -e test-alldeps --develop --notest
+    source .tox/test-alldeps/bin/activate
+
+Now, one can run any of the ``pytest`` commands below to only run tests that
+are relevant to the development one is doing.  If the development includes
+documentation, one could install the documentation dependencies in the same
```

Comment:
There is one catch, if I have things right, namely the `graphviz` dependency which must come either via conda or via a separate system-wide install. So for this `pip`-centric workflow you cannot imply all the docs dependencies are part of the "environment".

Now on Mac, doing a system install of graphviz requires either homebrew or macports.  I don't use either of those. So then for me that means making a base environment with conda and then a tox-generated virtualenv under that. This leads me to have a command line prompt like `(base) (test-alldeps)` which is a bit unseemly to me.

So despite having agreed the tox-virtualenv might be a good way forward, I have stopped using it in favor of just making a conda environment that has the right deps using #10343.

Reply:
I have removed the documentation part, as this could just make things harder than easier.